src/main/java/Duke.java

```diff
@@ -1,10 +1,325 @@
+    }
+
+    // Level-1: Exit Message triggered by "bye"
+    public String exit() {
+        this.activated = false;
+        return " Bye. Hope to see you again soon!";
+    }
+
+
+    // Level-2: adds new task to user
+    public String add(String text) {
+        Task nextTask = new Task(text);
+        this.storage.add(nextTask);
+        return "added: ".concat(text);
+    }
+    // Level-2: Lists all items in storage
+    public String list() {
+        String out = "Here are the tasks in your list:\n";
+        for (int i = 0; i < this.storage.size(); i++) {
+            if (i != 0) out = out.concat("\n");
+            out = out.concat((i+1) + ".").concat(this.storage.get(i).toString());
+        }
+        return out;
+    }
+
+    // Level-3: Mark items as done
+    public String done(int index) {
+        this.storage.get(index).setDone();
+        Task task = this.storage.get(index);
+        return "Nice! I've marked this task as done: \n  ".concat(task.toString());
+    }
+
+    // Level-4: setting of task urgency to "to-do"
+    public String todo(String description) {
+        Task task = new Todo(description,state.T);
+        this.storage.add(task);
+        return "Got it. I've added this task:\n  "
+                .concat(task.toString())
+                .concat("\nNow you have ".concat(this.storage.size() + "").concat(" tasks in the list."));
+    }
+
+    // Level-4: setting of task urgency to "deadline"
+    public String deadline(String description, String by) {
+        Task task = new Deadline(description,by,state.D);
+        this.storage.add(task);
+        return "Got it. I've added this task:\n  "
+                .concat(task.toString())
+                .concat("\nNow you have ".concat(this.storage.size() + "").concat(" tasks in the list."));
+    }
+
+    // Level-4: setting of task urgency to "event"
+    public String event(String description, String at) {
+        Task task = new Event(description,at,state.E);
+        this.storage.add(task);
+        return "Got it. I've added this task:\n  "
+                .concat(task.toString())
+                .concat("\nNow you have ".concat(this.storage.size() + "").concat(" tasks in the list."));
+    }
+
+    // Level-6: delete items from list
+    public String delete(int index) {
+        Task task = this.storage.get(index);
+        this.storage.remove(index);
+        return "Noted. I've removed this task:\n  "
+                .concat(task.toString())
+                .concat("\nNow you have ".concat(this.storage.size() + "").concat(" tasks in the list."));
+    }
+
+    // wraps all messages between line separators
+    public String messageWrapper(String text) {
+        return SEP_LINE.concat("\n").concat(text).concat("\n").concat(SEP_LINE);
+    }
+
+    // decodes input from user and passes that argument to a response builder
+    public int decoder(String userInput) {
+        int res;
+        switch (userInput) {
+            case "bye":
```

Comment:
For `switch-case` statements the indentation should be the same for `switch` and `case` as stated by the [coding standard](https://se-education.org/guides/conventions/java/intermediate.html)!